src/openproduct/utils/fields.py

```diff
@@ -12,3 +21,28 @@ def formfield(self, **kwargs):
         }
         defaults.update(kwargs)
         return super(ArrayField, self).formfield(**defaults)
+
+
+@extend_schema_field(OpenApiTypes.UUID)
+class UUIDRelatedField(serializers.RelatedField):
+    """
+    A read-write field that represents the target of the relationship
+    by a unique 'uuid' attribute.
+    """
+
+    default_error_messages = {
+        "does_not_exist": _("Object with uuid={value} does not exist."),
+        "invalid": _("Invalid value."),
+    }
+
+    def to_internal_value(self, data):
+        queryset = self.get_queryset()
+        try:
+            return queryset.get(**{"uuid": data})
+        except ObjectDoesNotExist:
+            self.fail("does_not_exist", value=smart_str(data))
+        except (TypeError, ValueError):
+            self.fail("invalid")
+
+    def to_representation(self, obj):
+        return attrgetter("uuid")(obj)
```

Comment:
can't you just use `getattr`?
```suggestion
        return getattr(obj, "uuid")
```

Reply:
changed, used it originally because SlugRelatedField does.